src/ripple/app/ledger/Ledger.cpp

```diff
@@ -1762,7 +1694,9 @@ void Ledger::updateFees ()
     }
 
     {
-        StaticScopedLockType sl (sPendingSaveLock);
+        // VFALCO Why not do this before calling getASNode?
+        std::lock_guard<
+            std::mutex> lock(mutex_);
```

Comment:
People shouldn't be changing `std::mutex` and more importantly we must avoid recursive mutexes.